### PR TITLE
Make it possible to use different Symbolicator pools

### PR DESCRIPTION
**Pull request by Swatinem**
Creates the main `Symbolicator` instance in the `symbolicate_event` task, and makes it possible to more easily route specific events to a specific symbolicator pool.

For this, introduce a `TaskKind` for `symbolicate_event` tasks.

This tries to contain the combinatorial explosion of different symbolication tasks behind a new type, and introduces new tasks/queues for js symbolication.

In summary:
- We have a new `SYMBOLICATOR_POOL_URLS` with `"lpq"`, `"js"` and `"default"`.
- We get two new queues / tasks: `symbolicate_js_event` and `symbolicate_js_event_low_priority`.

The idea is to have a separate LPQ queue for JS events, but still route those to the same symbolicator pool.

### CODE REVIEW

File: src/sentry/tasks/symbolication.py [hunk 1]
```diff
@@ -224,6 +221,22 @@ def record_symbolication_duration() -> float:
                     sentry_sdk.capture_exception(e)
         return symbolication_duration
 
+    # TODO: we should pick a different url / option, so we route symbolication request to
+    # different pools of symbolicators depending on `symbolication_function` and `is_low_priority`
```
Comment (loewenheim) [inline src/sentry/tasks/symbolication.py:225]: What is meant by "pick a different url / option" here?

File: src/sentry/lang/native/symbolicator.py [hunk 2]
```diff
@@ -28,18 +30,31 @@ def _task_id_cache_key_for_event(project_id, event_id):
     return f"symbolicator:{event_id}:{project_id}"
 
 
+@dataclass(frozen=True)
+class SymbolicatorTaskKind:
+    is_js: bool = False
+    is_low_priority: bool = False
+    is_reprocessing: bool = False
+
+    def with_low_priority(self, is_low_priority: bool) -> "SymbolicatorTaskKind":
```
Comment (loewenheim) [inline src/sentry/lang/native/symbolicator.py:39]: Why is the return type in quotes? Is it because it's the class being defined?

**Review by loewenheim — APPROVED** [review]

File: src/sentry/lang/native/symbolicator.py (hunk 2 above)
Comment (Swatinem) [reply src/sentry/lang/native/symbolicator.py:39]: Python can’t refer to the `Self` type directly in the statement that defines it, so using a string is the sanctioned workaround there.

File: src/sentry/lang/native/symbolicator.py [hunk 3]
```diff
@@ -28,18 +30,31 @@ def _task_id_cache_key_for_event(project_id, event_id):
...
+        return dataclasses.replace(self, is_low_priority=is_low_priority)
+
+
 class Symbolicator:
-    def __init__(self, project, event_id):
-        symbolicator_options = options.get("symbolicator.options")
-        base_url = symbolicator_options["url"].rstrip("/")
+    def __init__(self, task_kind: SymbolicatorTaskKind, project: Project, event_id: str):
+        URLS = settings.SYMBOLICATOR_POOL_URLS
+        pool = "default"
```
Comment (beezz) [inline src/sentry/lang/native/symbolicator.py:46]: nit: Preferably we have the string names of pools as constants (or enum) here and in settings.

**Review by beezz — CHANGES_REQUESTED** [review]

File: src/sentry/lang/native/symbolicator.py (hunk 3 above)
Comment (Swatinem) [reply src/sentry/lang/native/symbolicator.py]: I defined them in `symbolicator.py`, defining that in the settings seems weird, and does not play nice with typing either. Importing that into the settings would also introduce circular dependencies I guess, so I just added a comment there.

File: src/sentry/conf/server.py [hunk 4]
```diff
@@ -684,6 +684,11 @@ def SOCIAL_AUTH_DEFAULT_USERNAME():
     Queue(
         "events.symbolicate_event_low_priority", routing_key="events.symbolicate_event_low_priority"
     ),
+    Queue("events.symbolicate_js_event", routing_key="events.symbolicate_js_event"),
```
Comment (kamilogorek) [inline src/sentry/conf/server.py:687]: This can be removed now; Added in https://github.com/getsentry/sentry/pull/46936

**Review by beezz — APPROVED** [review]

**Pull request merged**